tests/tensor/random/test_basic.py

```diff
@@ -1272,3 +1272,25 @@ def test_pickle():
     a_unpkl = pickle.loads(a_pkl)
 
     assert a_unpkl.owner.op._props() == sample_a.owner.op._props()
+
+
+def test_rng_default_updates():
+    rng = shared(np.random.default_rng(42343))
+
+    v1 = normal(rng=rng)
+
+    assert rng.default_update == v1.owner.outputs[0]
+
+    with pytest.warns(Warning, match="default_update"):
+        normal(rng=rng)
+
+
+def test_rng_state_scan():
+    rng = shared(np.random.default_rng(42343))
+
+    def inner_fn(rng):
+        return normal(rng=rng)
+
+    out, updates = scan(inner_fn, non_sequences=[rng], n_steps=10)
+
+    assert len(set(out.eval())) == 10
```

Comment:
Any reason why you make use of `set` here? it seems `len` returns 10 without it.

Reply:
It was just a cheap way to make sure there were 10 unique elements.  It's definitely not a great way to do that, but with a fixed seed, it basically works.  Also, it's doesn't really _need_ to be `... == 10`; it could be `... > 1`.

Reply:
Ah, I see. `assert np.unique(out.eval()).size > 1` would work too but its a longer expression.